src/persistence/stock.rs

```diff
@@ -471,20 +462,19 @@ impl Inventory for Stock {
         self.consume_consignment(transfer, resolver, force)
     }
 
-    fn consume_anchor(
+    unsafe fn consume_anchor(
```

Comment:
Why unsafe?

Reply:
To prevent it from being called from any place other than the clearly defined single place.

Not a best practice, I know - but it is easier not to mistake/miss invalid calls in this way